plugins/woocommerce/includes/abstracts/abstract-wc-product.php

```diff
@@ -782,6 +782,10 @@ public function set_featured( $featured ) {
 	 */
 	public function set_catalog_visibility( $visibility ) {
 		$options = array_keys( wc_get_product_visibility_options() );
+
+		//Convert to lowercase
+		$visibility = strtolower( $visibility );
```

Comment:
Here we need to be a bit more careful: given `wc_get_product_visibility_options()` calls a filter, plugins can add their own visibility options, and we can't be sure those correspond to lowercase keys, like the default ones. That is, the options returned by `wc_get_product_visibility_options()` might be case sensitive.

As such, I think what we want to do here is first check if `$visibility` is in `$options`. If it is, we use it verbatim. If it isn't, _but_ `strtolower( $visibility )` is, we use the lowercase version.

That would mean we respect case sensitivity but at the same time makes things easier, especially for the default visibility options.

Reply:
@jorgeatorres No problem with the delay. I understand WooCommerce has many other priorities going on. Here, Can we first check the visibility options with the original case? If isn't a match then we can do a check-in with strtolower? I would be happy to help with this ticket.

WDYT?

Reply:
Hi @ajayghaghretiya-multidots!

Thanks for your understanding. The approach you've suggested sounds reasonable.
Feel free to push the relevant changes and ask for a re-review from me. Thanks again!